src/port1.0/portconfigure.tcl

```diff
@@ -513,87 +590,472 @@ proc portconfigure::configure_get_default_compiler {} {
+        lappend min_values 4.4
+    }
+    if {${compiler.thread_local_storage} && ![thread_local_storage_possible gcc]} {
+        lappend min_values 9999.0
+    }
+    return [lindex [lreverse [lsort -command vercmp ${min_values}]] 0]
+}
+#
+proc portconfigure::g95_ok {} {
+    global compiler.openmp_version
+    if {${compiler.openmp_version} eq ""} {
+        return yes
+    } else {
+        return no
+    }
+}
+# utility procedure: get Xcode compilers
+proc portconfigure::get_xcode_compilers {} {
+    global xcodeversion
+    # https://developer.apple.com/library/content/releasenotes/DeveloperTools/RN-Xcode/Chapters/Introduction.html
+    # https://developer.apple.com/library/content/documentation/CompilerTools/Conceptual/LLVMCompilerOverview/index.html
+    # Xcode 3.2 relase notes (Link?)
+    # About Xcode 3.1 Tools (about_xcode_tools_3.1.pdf, Link?)
+    # About Xcode 3.2 (about_xcode_3.2.pdf, Link?)
+    #
+    # Xcode 5 does not support use of the LLVM-GCC compiler and the GDB debugger.
+    # From Xcode 4.2, Clang is the default compiler for Mac OS X.
+    # llvm-gcc4.2 is now the default system compiler in Xcode 4.
+    # The LLVM compiler is the next-generation compiler, introduced in Xcode 3.2
+    # GCC 4.0 has been removed from Xcode 4.
+    #
+    # attempt to include all available compilers except gcc-3*
+    # attempt to have the default compilers first
+    if {[vercmp ${xcodeversion} 5] >= 0} {
+        set compilers {clang}
+    } elseif {[vercmp ${xcodeversion} 4.3] >= 0} {
+        set compilers {clang llvm-gcc-4.2}
+    } elseif {[vercmp ${xcodeversion} 4.2] >= 0} {
+        # llvm-gcc is more reliable
+        # see https://github.com/macports/macports-base/commit/10d62cb51b1f0f9703a873173bac468eee69d01a
+        set compilers {llvm-gcc-4.2 clang}
+    } elseif {[vercmp ${xcodeversion} 4.0] >= 0} {
+        set compilers {llvm-gcc-4.2 clang gcc-4.2}
+    } else {
+        # Legacy Cases
+        if {[string match *10.4u* [option configure.sdkroot]]} {
+            # from Xcode 3.2 release notes:
+            #    GCC 4.2 cannot be used with the Mac OS X 10.4u SDK.
+            #    If you want to build targets using the 10.4u SDK on Xcode 3.2, you must set the Compiler Version to GCC 4.0
+            set compilers {gcc-4.0}
+        } else {
+            if {[vercmp ${xcodeversion} 3.2] >= 0} {
+                # from about_xcode_3.2.pdf:
+                #    GCC 4.2 is the primary system compiler for the 10.6 SDK
+                # clang does *not* provide clang++, but configure.cxx will fall back to llvm-g++-4.2
+                set compilers {gcc-4.2 llvm-gcc-4.2 clang gcc-4.0}
+            } elseif {[vercmp ${xcodeversion} 3.1] >= 0} {
+                # from about_xcode_tools_3.1.pdf:
+                #     GCC 4.2 & LLVM GCC 4.2 optional compilers
+                # assume they exist
+                set compilers {gcc-4.2 llvm-gcc-4.2 apple-gcc-4.2 gcc-4.0}
```

Comment:
What is the purpose of get_xcode_compilers()?  It seems like it's not clearly defined since it is in some cases returning only the compilers provided by Xcode, and in other cases it is also returning MacPorts-provided compilers that are preferred over the Xcode-provided compilers.